client/src/views/mapping/components/inputColumn/addJoinButton.tsx

```diff
@@ -0,0 +1,36 @@
+import { Button } from "@blueprintjs/core";
+
+import * as React from "react";
+
+import { Mutation } from "react-apollo";
+
+const createJoinAndUpdateInputColumn = require("../../../../graphql/mutations/createJoinAndUpdateInputColumn.graphql");
+
+interface IProps {
+  column: any;
+}
+
+const InputColumnAddJoin = ({ column }: IProps) => (
+  <Mutation mutation={createJoinAndUpdateInputColumn}>
```

Comment:
Although I am not doing it yet, I think I'll stop using `Query` and `Mutation` tags, and rather opt for `useQuery` or `useMutation` from `@apollo/react-hooks`: https://www.apollographql.com/docs/react/api/react-hooks/#usequery

Imho, it prevents nesting "service logic" and "components" and leads to cleaner code :slightly_smiling_face: 
Don't necessarily change this component now though...